### PR TITLE
ci: add PR labeler

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: NA

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

File: .github/pr_labeler.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  - buf.work.yaml
+  - .mergify.yml
+  - .golangci.yml
+  - mlc_config.json
+  - sonar-project.properties
+"C:Build":
+  - Makefile
+  - Dockerfile
+  - scripts/*
+"C:Testing":
```
Comment (insumity) [inline .github/pr_labeler.yml:24]: Is the `C:Testing` the label applied when change under `app/**/*`, `cmd/**/*` etc. If yes, why introduce this label for changes under `app` and `cmd`? Also, would it be possible to add this label if modification in a `_test` file.

**Review by insumity — APPROVED** [review]
LGTM

File: .github/pr_labeler.yml (hunk 1 above)
Comment (mpoke) [reply .github/pr_labeler.yml:24]: Both app/* and cmd/* are only used for testing. Every chain will have its own. Maybe we can have a better split, maybe with a better label. I'll let @p-offtermatt and @MSalopek to suggest something. 

All PRs that change the production code should also change the _test files as they will have unit tests. I'm not sure how valuable is to add a special label for that.

**Review by smarshall-spitzbart — APPROVED** [review]
Seems we could also just look at what files are changed for a PR and determine these mental notes ourselves, but overall not opposed to these tags. LGTM

File: .github/pr_labeler.yml (hunk 1 above)
Comment (p-offtermatt) [reply .github/pr_labeler.yml:24]: I don't think I have an opinion on that yet, I would suggest we see how these labels feel and we can adjust later.

**Pull request merged**